packages/server/api/src/app/store-entry/store-entry.service.ts

```diff
@@ -62,4 +62,34 @@ export const storeEntryService = {
       key,
     });
   },
+
+  async list({
+    projectId,
+    prefix,
+    filterRegex,
+  }: {
+    projectId: ProjectId;
+    prefix: string;
+    filterRegex?: string;
+  }): Promise<Array<{ key: string; value: unknown }>> {
+    const query = storeEntryRepo()
+      .createQueryBuilder('storeEntry')
+      .where('storeEntry.projectId = :projectId', { projectId });
+
+    let keyExpression = 'storeEntry.key';
+
+    const escapedPrefix = prefix.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    query.andWhere('storeEntry.key LIKE :prefix', { prefix: `${prefix}%` });
+    if (filterRegex) {
+      keyExpression = `REGEXP_REPLACE(${keyExpression}, '^${escapedPrefix}', '', 'g')`;
+    }
+
+    if (filterRegex) {
+      query.andWhere(`${keyExpression} ~ :filterRegex`, { filterRegex });
+    }
```

Comment:
```suggestion
    if (filterRegex) {
      keyExpression = `REGEXP_REPLACE(${keyExpression}, '^${escapedPrefix}', '', 'g')`;
      query.andWhere(`${keyExpression} ~ :filterRegex`, { filterRegex });
    }
```

I don't think we need the keyExpression variable